pkg/daemon/update.go

```diff
@@ -474,6 +598,96 @@ func calculatePostConfigChangeActionFromFileDiffs(diffFileSet []string) (actions
 	return
 }
 
+// calculatePostConfigChangeNodeDisruptionActionFromMCDiffs takes action based on the cluster's Node disruption policies.
+func calculatePostConfigChangeNodeDisruptionActionFromMCDiffs(diffSSH bool, diffFileSet, diffUnitSet []string, clusterPolicies opv1.NodeDisruptionPolicyClusterStatus) []opv1.NodeDisruptionPolicyStatusAction {
+	actions := []opv1.NodeDisruptionPolicyStatusAction{}
+
+	// Step through all file based policies, and build out the actions object
+	for _, diffPath := range diffFileSet {
+		pathFound := false
+		for _, policyFile := range clusterPolicies.Files {
+			klog.V(4).Infof("comparing policy path %s to diff path %s", policyFile.Path, diffPath)
+			if policyFile.Path == diffPath {
+				klog.Infof("NodeDisruptionPolicy found for diff file %s!", diffPath)
```

Comment:
nit: Perhaps we should exclude `!` in the end since it can be confusing that it is part of filename.

Reply:
ack, will fix alongside anything QE might find on next pass. Thanks!